executor/src/witgen/query_processor.rs

```diff
@@ -79,16 +79,16 @@ where
                 let v = self.evaluate_expression(scrutinee, rows)?;
                 let expr = arms
                     .iter()
-                    .find_map(|MatchArm { pattern, value }| {
-                        (match pattern {
-                            MatchPattern::CatchAll => true,
-                            MatchPattern::Pattern(pattern) => {
-                                self.evaluate_expression(pattern, rows) == Ok(v)
+                    .find_map(|MatchArm { pattern, value }| match pattern {
+                        MatchPattern::CatchAll => Some(Ok(value)),
+                        MatchPattern::Pattern(pattern) => {
+                            match self.evaluate_expression(pattern, rows) {
+                                Ok(p) => (p == v).then_some(Ok(value)),
+                                Err(e) => Some(Err(e)),
                             }
-                        })
-                        .then_some(value)
+                        }
                     })
-                    .ok_or(IncompleteCause::NoMatchArmFound)?;
+                    .ok_or(IncompleteCause::NoMatchArmFound)??;
```

Comment:
Hmm, a bit confusing with the double quotation mark, but I don't see an easy improvement (neither does ChatGPT...)